client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,162 @@
+	const jetpackRegeneratingPostByEmail = siteIsJetpack && emailAddress === 'regenerate';
+	const jetpackPostByEmailIsActive = !! useSelector( ( state ) =>
+		isJetpackModuleActive( state, selectedSiteId, moduleSlug )
+	);
+	const siteInDevMode = useSelector( ( state ) =>
+		isJetpackSiteInDevelopmentMode( state, selectedSiteId )
+	);
+	const moduleUnavailableInDevMode = useSelector( ( state ) =>
+		isJetpackModuleUnavailableInDevelopmentMode( state, selectedSiteId, moduleSlug )
+	);
+	const moduleUnavailable = siteInDevMode && moduleUnavailableInDevMode;
+
+	const { data: simpleSitePostByEmailSettings } = useGetPostByEmail( selectedSiteId );
+	const { mutate: simpleSiteTogglePostByEmail, isPending: isSimpleSitePendingToggle } =
+		useTogglePostByEmailMutation( selectedSiteId );
+	const { mutate: simpleSiteRegeneratePostByEmail, isPending: isSimpleSitePendingRegenerate } =
+		useRegeneratePostByEmailMutation( selectedSiteId );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleToggleForSimpleSite = ( checked: boolean ) => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		simpleSiteTogglePostByEmail( checked, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Settings saved successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem saving your changes.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		if ( siteIsJetpack ) {
+			dispatch( jetpackRegeneratePostByEmail( selectedSiteId ) );
+			return;
+		}
+
+		simpleSiteRegeneratePostByEmail( undefined, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Address regenerated successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem regenerating the address.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	let email;
+	if ( siteIsJetpack ) {
+		email = emailAddress === 'regenerate' ? '' : emailAddress;
+	} else {
+		email = simpleSitePostByEmailSettings?.email;
+	}
+
+	const isActive = siteIsJetpack
+		? jetpackPostByEmailIsActive
+		: simpleSitePostByEmailSettings?.isEnabled;
+	const isJetpackDisabled = jetpackRegeneratingPostByEmail || moduleUnavailable;
+	const isSimpleSiteDisabled = isSimpleSitePendingToggle || isSimpleSitePendingRegenerate;
+	const isDisabledControls = ! isActive || isJetpackDisabled || isSimpleSiteDisabled;
+
+	const toggleLabel = translate( 'Publish posts by sending an email' );
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						! siteIsJetpack
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ siteIsJetpack }
```

Comment:
We used to look at `siteIsAtomic` here and the logic was flipped (Simple sites linked to jetpack.com and Atomic sites to wordpress.com).

The flipped logic was obviously wrong, but I also don't see a reason to look at `siteIsAtomic` instead of `siteIsJetpack`.